error: future cannot be sent between threads safely
  --> $DIR/issue-68112.rs:34:5
   |
LL | fn require_send(_: impl Send) {}
   |    ------------         ---- required by this bound in `require_send`
...
LL |     require_send(send_fut);
   |     ^^^^^^^^^^^^ future created by async block is not `Send`
   |
   = help: the trait `std::marker::Sync` is not implemented for `std::cell::RefCell<i32>`
note: future is not `Send` as this value is used in an await
  --> $DIR/issue-68112.rs:31:17
   |
LL |         let non_send_fut = make_non_send_future1();
   |             ------------ created here
LL |         let _ = non_send_fut.await;
   |                 ^^^^^^^^^^^^ await occurs here

error[E0277]: `std::cell::RefCell<i32>` cannot be shared between threads safely
  --> $DIR/issue-68112.rs:49:5
   |
LL | fn require_send(_: impl Send) {}
   |    ------------         ---- required by this bound in `require_send`
...
LL |     require_send(send_fut);
   |     ^^^^^^^^^^^^ `std::cell::RefCell<i32>` cannot be shared between threads safely
   |
   = help: the trait `std::marker::Sync` is not implemented for `std::cell::RefCell<i32>`
   = note: required because of the requirements on the impl of `std::marker::Send` for `std::sync::Arc<std::cell::RefCell<i32>>`
   = note: required because it appears within the type `[static generator@$DIR/issue-68112.rs:38:31: 38:36 t:std::sync::Arc<std::cell::RefCell<i32>> {}]`
   = note: required because it appears within the type `std::future::from_generator::GenFuture<[static generator@$DIR/issue-68112.rs:38:31: 38:36 t:std::sync::Arc<std::cell::RefCell<i32>> {}]>`
   = note: required because it appears within the type `impl std::future::Future`
   = note: required because it appears within the type `impl std::future::Future`
   = note: required because it appears within the type `impl std::future::Future`
   = note: required because it appears within the type `{std::future::ResumeTy, impl std::future::Future, (), i32, Ready<i32>}`
   = note: required because it appears within the type `[static generator@$DIR/issue-68112.rs:44:26: 48:6 {std::future::ResumeTy, impl std::future::Future, (), i32, Ready<i32>}]`
   = note: required because it appears within the type `std::future::from_generator::GenFuture<[static generator@$DIR/issue-68112.rs:44:26: 48:6 {std::future::ResumeTy, impl std::future::Future, (), i32, Ready<i32>}]>`
   = note: required because it appears within the type `impl std::future::Future`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
